include/libtorrent/bencode.hpp

```diff
@@ -163,7 +163,7 @@ namespace libtorrent
 				ret += 2;
 				break;
 			case entry::string_t:
-				ret += write_integer(out, e.string().length());
+				ret += write_integer(out, entry::integer_type(e.string().length()));
```

Comment:
what would you think of giving ``write_integer()`` a similar treatment as the ``write_int*()``? i.e. have built-in conversions from other integral types